x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java

```diff
@@ -90,6 +97,35 @@ public ConnectorSyncJobIndexService(Client client) {
         this.clientWithOrigin = new OriginSettingClient(client, CONNECTORS_ORIGIN);
     }
 
+    /**
+     * Returns the {@link SystemIndexDescriptor} for the Connector system index.
+     *
+     * @return The {@link SystemIndexDescriptor} for the Connector system index.
+     */
+    public static SystemIndexDescriptor getSystemIndexDescriptor() {
+        PutIndexTemplateRequest request = new PutIndexTemplateRequest();
+        String templateSource = TemplateUtils.loadTemplate(
+            "/elastic-connectors-sync-jobs.json",
+            Version.CURRENT.toString(),
+            CONNECTOR_SYNC_JOB_MAPPING_VERSION_VARIABLE,
+            Map.of(CONNECTOR_SYNC_JOB_MAPPING_MANAGED_VERSION_VARIABLE, Integer.toString(CONNECTOR_SYNC_JOB_INDEX_VERSION))
+        );
+        request.source(templateSource, XContentType.JSON);
+
+        return SystemIndexDescriptor.builder()
```

Comment:
Same comment here on building the system index descriptor as in `ConnectorIndexService`.

Reply:
I've cleaned up the consts here too